### PR TITLE
[DOXIA-590] implement hidden rows

### PR DESCRIPTION
**Pull request by feckertson**

### CODE REVIEW

**Comment by feckertson** [comment]
Decides whether to toggle the even/odd table row parity for zebra striping based on whether the 'hidden' class is being applied.

Also adds unit tests for setting an externalLink and a class string simultaneously.

File: doxia-core/src/main/java/org/apache/maven/doxia/sink/impl/Xhtml5BaseSink.java [hunk 1]
```diff
@@ -61,17 +61,20 @@
...
     /** An indication on if we're inside a head. */
     private boolean headFlag;
 
     /** Keep track of the main and div tags for content events. */
-    protected Stack<Tag> contentStack = new Stack<>();
+    protected Stack<Tag> contentStack = new Stack<Tag>();
 
     /** Keep track of the closing tags for inline events. */
-    protected Stack<List<Tag>> inlineStack = new Stack<>();
+    protected Stack<List<Tag>> inlineStack = new Stack<List<Tag>>();
```
Comment (michael-o) [inline doxia-core/src/main/java/org/apache/maven/doxia/sink/impl/Xhtml5BaseSink.java:77]: Why?

File: doxia-core/src/main/java/org/apache/maven/doxia/sink/impl/Xhtml5BaseSink.java [hunk 2]
```diff
@@ -61,17 +61,20 @@
...
+
     /** Used to collect text events mainly for the head events. */
     private StringBuffer textBuffer = new StringBuffer();
 
     /** An indication on if we're inside a head. */
     private boolean headFlag;
 
     /** Keep track of the main and div tags for content events. */
-    protected Stack<Tag> contentStack = new Stack<>();
+    protected Stack<Tag> contentStack = new Stack<Tag>();
```
Comment (michael-o) [inline doxia-core/src/main/java/org/apache/maven/doxia/sink/impl/Xhtml5BaseSink.java:74]: Why?

File: doxia-core/src/main/java/org/apache/maven/doxia/sink/impl/Xhtml5BaseSink.java [hunk 3]
```diff
@@ -136,13 +139,13 @@ public Xhtml5BaseSink( Writer out )
...
-        this.tableCaptionWriterStack = new LinkedList<>();
-        this.tableCaptionXMLWriterStack = new LinkedList<>();
-        this.tableCaptionStack = new LinkedList<>();
+        this.cellJustifStack = new LinkedList<int[]>();
+        this.isCellJustifStack = new LinkedList<Boolean>();
+        this.cellCountStack = new LinkedList<Integer>();
+        this.tableContentWriterStack = new LinkedList<StringWriter>();
+        this.tableCaptionWriterStack = new LinkedList<StringWriter>();
+        this.tableCaptionXMLWriterStack = new LinkedList<PrettyPrintXMLWriter>();
+        this.tableCaptionStack = new LinkedList<String>();
```
Comment (michael-o) [inline doxia-core/src/main/java/org/apache/maven/doxia/sink/impl/Xhtml5BaseSink.java:148]: Why?

**Review by michael-o — COMMENTED** [review]
Why do I see the diamond changes?

**Comment by michael-o** [comment]
Will check in the next couple of dazs.

**Comment by feckertson** [comment]
Apologies for not checking the diff before creating the PR.  The apache format introduced unintended changes/noise. The standard format should be applied, but in a PR which only does that.

I couldn't find standard eclipse settings (code cleanup) for this org/project to avoid the diamond changes?  Are there any?

**Review by michael-o — CHANGES_REQUESTED** [review]
I don't see handling for `onSection`.

File: doxia-core/src/test/java/org/apache/maven/doxia/sink/impl/Xhtml5BaseSinkTest.java [hunk 4]
```diff
@@ -1086,15 +1086,86 @@ public void testTableRow()
...
+            sink.tableRow_();
+            sink.tableRows_();
+            sink.table_();
+        }
+        finally
+        {
+            sink.close();
+        }
+
+        String nl = System.lineSeparator();
```
Comment (michael-o) [inline doxia-core/src/test/java/org/apache/maven/doxia/sink/impl/Xhtml5BaseSinkTest.java:1102]: Use provided `LS`

File: doxia-core/src/test/java/org/apache/maven/doxia/sink/impl/Xhtml5BaseSinkTest.java [hunk 5]
```diff
@@ -1086,15 +1086,86 @@ public void testTableRow()
...
+            sink.tableRow();
+            sink.tableRow_();
         }
         finally
         {
             sink.close();
         }
 
-        assertEquals( "<tr style=\"bold\" class=\"a\"></tr>", writer.toString() );
+        String nl = System.lineSeparator();
```
Comment (michael-o) [inline doxia-core/src/test/java/org/apache/maven/doxia/sink/impl/Xhtml5BaseSinkTest.java:1154]: Use provided `LS`

**Review by michael-o — CHANGES_REQUESTED** [review]
A few nits. Otherwise your change looks reasonable. @hboutemy Any objections?

**Comment by feckertson** [comment]
>  I don't see handling for onSection.

Please explain. I see no issues with the way onSection is coded (unless the concern is that onSection does nothing if the level is greater than 5).

**Comment by michael-o** [comment]
> > I don't see handling for onSection.
> 
> Please explain. I see no issues with the way onSection is coded (unless the concern is that onSection does nothing if the level is greater than 5).

See here: https://github.com/apache/maven-doxia/blob/8ef6a26a35946385ad55193a63e0180a5d5320ec/doxia-core/src/main/java/org/apache/maven/doxia/sink/impl/XhtmlBaseSink.java#L472-L484

**Comment by michael-o** [comment]
Merged. I have ignored `onSection()` because it will be removed in 2.0 anyway.

**Pull request closed without merging**